src/measure_theory/vector_measure.lean

```diff
@@ -427,4 +427,94 @@ end
 
 end measure
 
+namespace vector_measure
+
+section
+
+variables [measurable_space β]
+variables {M : Type*} [add_comm_monoid M] [topological_space M]
+
+/-- The pullback of a vector measure along a measurable function. -/
+def map (v : vector_measure α M) (f : α → β) (hf : measurable f) :
+  vector_measure β M :=
+{ measure_of' := λ s, if measurable_set s then v (f ⁻¹' s) else 0,
+  empty' := by simp,
+  not_measurable' := λ i hi, if_neg hi,
+  m_Union' :=
+  begin
+    intros g hg₁ hg₂,
+    convert v.m_Union (λ i, hf (hg₁ i)) (λ i j hij x hx, hg₂ i j hij hx),
+    { ext i, rw if_pos (hg₁ i) },
+    { rw [preimage_Union, if_pos (measurable_set.Union hg₁)] }
+  end }
+
+lemma map_apply {f : α → β} (hf : measurable f)
+  (v : vector_measure α M) {s : set β} (hs : measurable_set s) :
+  v.map f hf s = v (f ⁻¹' s) :=
+if_pos hs
+
+lemma map_id (v : vector_measure α M) : v.map id (measurable_id) = v :=
+ext (λ i hi, if_pos hi)
+
+/-- The restriction of a vector measure on some measurable set-/
+def restrict (v : vector_measure α M) (i : set α) (hi : measurable_set i) :
```

Comment:
can you also register the group morphism version? And also not require measurability in the definition (but saying that you get `0` when the set is not measurable), to follow the nonnegative measures API?